File: CS/Form1.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraRichEdit.API.Native;
using DevExpress.XtraRichEdit.Utils;

namespace RichEditHighlightCurrentRow {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();

            richEditControl1.LoadDocument(System.IO.Directory.GetCurrentDirectory() + @"\..\..\MovieRentals.docx");
        }

        private void richEditControl1_Paint(object sender, PaintEventArgs e) {
            DocumentRange range = richEditControl1.Document.Selection;
            Rectangle rect = richEditControl1.GetBoundsFromPosition(range.Start);

            rect = new Rectangle(
                0,
                (int)Units.DocumentsToPixels(rect.Top, richEditControl1.DpiX),
                richEditControl1.ClientSize.Width,
                (int)Units.DocumentsToPixels(rect.Height, richEditControl1.DpiX));

            Brush brush = new SolidBrush(Color.FromArgb(100, Color.Red));
            e.Graphics.FillRectangle(brush, rect);
        }

        private void richEditControl1_SelectionChanged(object sender, EventArgs e) {
            richEditControl1.Refresh();
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraRichEdit.API.Native
Imports DevExpress.XtraRichEdit.Utils

Namespace RichEditHighlightCurrentRow
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()

			richEditControl1.LoadDocument(System.IO.Directory.GetCurrentDirectory() & "\..\..\MovieRentals.docx")
		End Sub

		Private Sub richEditControl1_Paint(ByVal sender As Object, ByVal e As PaintEventArgs) Handles richEditControl1.Paint
			Dim range As DocumentRange = richEditControl1.Document.Selection
			Dim rect As Rectangle = richEditControl1.GetBoundsFromPosition(range.Start)

			rect = New Rectangle(0, CInt(Fix(Units.DocumentsToPixels(rect.Top, richEditControl1.DpiX))), richEditControl1.ClientSize.Width, CInt(Fix(Units.DocumentsToPixels(rect.Height, richEditControl1.DpiX))))

			Dim brush As Brush = New SolidBrush(Color.FromArgb(100, Color.Red))
			e.Graphics.FillRectangle(brush, rect)
		End Sub

		Private Sub richEditControl1_SelectionChanged(ByVal sender As Object, ByVal e As EventArgs) Handles richEditControl1.SelectionChanged
			richEditControl1.Refresh()
		End Sub
	End Class
End Namespace